x-pack/filebeat/input/s3/s3_integration_test.go

```diff
@@ -0,0 +1,417 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/sqsiface"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/tests/resources"
+	awscommon "github.com/elastic/beats/x-pack/libbeat/common/aws"
+)
+
+const (
+	fileName          = "sample1.txt"
+	fileDir           = "./ftest/" + fileName
```

Comment:
Use `filepath.Join`, and this doesn't look like a directory, what about this?
```suggestion
	filePath          = filepath.Join("ftest", fileName)
```